core-java-modules/core-java-nio-2/src/main/java/com/baeldung/lock/FileLocks.java

```diff
@@ -0,0 +1,239 @@
+package com.baeldung.lock;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.channels.NonReadableChannelException;
+import java.nio.channels.NonWritableChannelException;
+import java.nio.channels.OverlappingFileLockException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Charsets;
+
+
+public class FileLocks {
+	
+	private static Logger log = LoggerFactory.getLogger(FileLocks.class);
+
+	// Write locks
+	/**
+	 * Trying to get an exclusive lock on a read-only FileChannel won't work.
+	 */
+	static void getExclusiveLockFromInputStream() throws IOException, NonWritableChannelException {
+		Path path = Files.createTempFile("foo", "txt");
+		try (FileInputStream fis = new FileInputStream(path.toFile()); FileLock lock = fis.getChannel().lock()) {
+			log.debug("This won't happen");
+		} catch (NonWritableChannelException e) {
+			log.error(
+					"The channel obtained through a FileInputStream isn't writable. "
```

Comment:
Why do we need to concat the string here?

Reply:
No, this was because I was not using the right formatting. I used concatenation to break the statement into 2 lines because, usually, I limit the width of lines in my style. I'll fix this.